lib/sles4sap/console_redirection.pm

```diff
@@ -0,0 +1,170 @@
+    # ID serves as identification of the 'base' VM to return to.
+    set_var('WORKER_VM_ID', script_output 'cat /etc/machine-id');
+}
+
+=head2 set_serial_term_prompt
+
+    set_serial_term_prompt();
+
+Set expected serial prompt according to user which is currently active.
+This changes global setting $testapi::distri->{serial_term_prompt} which is important for calls like wait_for_serial.
+
+=cut
+
+sub set_serial_term_prompt {
+    $testapi::distri->{serial_term_prompt} = '';    # reset previous prompt first
+    my $current_user = script_output('whoami');
+    croak 'SSH user not defined and/or "whoami" script failed.' unless defined($current_user);
+    $testapi::distri->{serial_term_prompt} = ($current_user eq 'root' ? '# ' : '> ');
+}
+
+=head2 connect_target_to_serial
+
+    connect_target_to_serial([ssh_user=>ssh_user, target_ip=>$target_ip]);
+
+B<ssh_user>: Login user - default value is defined by OpenQA parameter REDIRECT_TARGET_USER
+B<target_ip>: Target host IP - default value is defined by OpenQA parameter REDIRECT_TARGET_IP
+
+Establishes ssh connection to target and redirects serial output to serial concole on worker VM.
+This allows OpenQA access to command return codes and output for evaulation by standard API call.
+
+=cut
+
+sub connect_target_to_serial {
+    my (%args) = @_;
+    $args{'target_ip'} //= get_required_var('REDIRECT_TARGET_IP');
+    $args{'ssh_user'} //= get_required_var('REDIRECT_TARGET_USER');
+
+    croak 'Missing "ssh_user" argument' unless $args{'ssh_user'};
+    croak 'Missing "redirect_ip" argument' unless $args{'target_ip'};
+    croak "OpenQA variable WORKER_VM_ID undefined. Run 'redirection_init()' first" unless get_var('WORKER_VM_ID');
+    croak "IP address '$args{'target_ip'}' is not valid." unless grep(/^$RE{net}{IPv4}$/, $args{'target_ip'});
+    croak 'Global variable "$searialdev" undefined' unless $serialdev;
+    croak "Console is already redirected to:" . script_output('hostname') if check_serial_redirection();
+
+    enter_cmd "ssh $ssh_opt $args{'ssh_user'}\@$args{'target_ip'} 2>&1 | tee -a /dev/$serialdev";
+    handle_login_prompt($args{'ssh_user'});
+    check_serial_redirection();
+    record_info('Redirect ON', "Serial redirection established to: $args{'target_ip'}");
+}
+
+=head2 disconnect_target_from_serial
+
+    disconnect_target_from_serial([worker_machine_id=$worker_machine_id]);
+
+B<worker_machine_id>: Target host IP - default value is defined by OpenQA parameter WORKER_VM_ID from redirect_init()
+
+Disconnects target from serial console by typing 'exit' command until host machine ID matches ID of the worker VM.
+
+=cut
+
+sub disconnect_target_from_serial {
+    my (%args) = @_;
+    $args{worker_machine_id} //= get_required_var('WORKER_VM_ID');
+    set_serial_term_prompt();
+    my $serial_redirection_status = check_serial_redirection(worker_machine_id => $args{worker_machine_id});
+    while ($serial_redirection_status ne 0) {
```

Comment:
Better use `!=` and not `ne` when comparing numbers.

Reply:
fixed